src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp

```diff
@@ -255,8 +255,8 @@ bool ov::pass::MOCTransformations::run_on_model(const std::shared_ptr<ngraph::Fu
         for (auto&& param : f->get_parameters()) {
             param->set_partial_shape(input_shapes.at(param.get()));
         }
-        f->validate_nodes_and_infer_types();
     }
+    f->validate_nodes_and_infer_types();
```

Comment:
could you add c++ test?
with Add->Add->Add + LinOp fusion transformation and check that we can set the names

Reply:
Done.